core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -96,7 +96,7 @@ final class KafkaMetadataLog private (
   }
 
   private def handleAndConvertLogAppendInfo(appendInfo: internals.log.LogAppendInfo): LogAppendInfo = {
-    if (appendInfo.firstOffset != UnifiedLog.UnknownOffset)
+    if (appendInfo.numMessages() > 0)
```

Comment:
The calc for `numMessages` is below:
```
    public long numMessages() {
        if (firstOffset >= 0 && lastOffset >= 0) {
            return lastOffset - firstOffset + 1;
        }
        return 0;
    }
```

I think this captures a similar intent to checking whether the first offset is `-1` or not.